### PR TITLE
Feat/auto increment run

### LINKED ISSUE #402: copy_mod() (or maybe change copy_model_from())
This is a nice helper to take a model and create a new model from it that automatically uses the next available integer in the directory, and calls [`update_run_number()`](https://github.com/metrumresearchgroup/bbr/issues/400) on the resulting control stream.

# Discussion points

* **should this be it's own function or should we rework it to be part of `copy_model_from()`**, such that you could specify some arguments to make it behave in this way?

# Proposed code

```r

copy_mod <- function(.parent_mod, ...){
  ## grab model directory path
  dir <- get_output_dir(.parent_mod) %>% dirname()
  # dir <- build_path_from_model(.parent_mod) %>% dirname()
  ## find next integer
  child_mod <- get_next_integer(dir, ...)
  ## copy model using default settings
  copy_model_from(.parent_mod=.parent_mod, .new_model=child_mod, 
                  .inherit_tags = T, ...) %>%
    update_run_number()
}


## return next largest integer for CTL files in the model directory
get_next_integer <- function(.dir, .padding = 3){
  mod.n <- list.files(.dir, pattern="ctl") %>% 
    str_extract_all("\\d+") %>% 
    unlist %>% 
    as.integer %>% 
    max + 1
  
  if(.padding > 0) mod.n <- stringr::str_pad(mod.n, .padding, pad = "0")
  
  return(mod.n)
}

```

### PR DESCRIPTION
**Pull request by seth127**
Closes #402

### CODE REVIEW

**Review by timwaterhouse — COMMENTED** [review]
Looks good, thanks

File: R/copy-model-from.R [hunk 1]
```diff
@@ -212,10 +221,53 @@ copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite,
...
+
+#' Check directory for model files with integer names
+#' and return the next integer. Used by copy_model_from().
+#' @importFrom stringr str_extract_all str_pad
+#' @keywords internal
+get_next_integer <- function(.parent_mod){
+  .dir <- dirname(get_model_path(.parent_mod))
+  .ext <- tools::file_ext(get_model_path(.parent_mod))
+
+  # find largest model in same dir with largest integer name
```
Comment (kyleam) [inline R/copy-model-from.R:251]: `s/largest model/largest/`

"largest model" tripped me up given the "largest integer name" that follows, and seems redundant at best.

File: R/copy-model-from.R [hunk 2]
```diff
@@ -212,10 +221,53 @@ copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite,
...
+  mods <- fs::dir_ls(.dir, regexp=glue("\\.{.ext}$")) %>%
+    basename() %>%
+    tools::file_path_sans_ext()
+
+  mods_int <- suppressSpecificWarning(as.integer(mods), .regexpr = "NAs introduced by coercion")
+  if (all(is.na(mods_int))) {
+    stop(paste(
+      glue("There are no models in {.dir} with integer names, so cannot increment to `.new_model` to next integer."),
```
Comment (kyleam) [inline R/copy-model-from.R:259]: drop "to" in front of `.new_model`?

File: tests/testthat/test-copy-model-from.R [hunk 3]
```diff
@@ -155,3 +155,46 @@ test_that("copy_model_from() supports `.new_model` containing a period [BBR-CMF-
...
+  new_mod2 <- copy_model_from(new_mod1)
+  expect_equal(
+    as.integer(get_model_id(new_mod1)) + 1,
+    as.integer(get_model_id(new_mod2))
+  )
+  expect_equal(!!nchar(get_model_id(new_mod2)), !!nchar(new_id))
+})
+
+
+test_that("copy_model_from(.new_model=NULL) increments to next integer highest [BBR-CMF-007]", {
```
Comment (kyleam) [inline tests/testthat/test-copy-model-from.R:173]: `s/integer highest/highest integer/`?

File: tests/testthat/test-copy-model-from.R [hunk 4]
```diff
@@ -155,3 +155,46 @@ test_that("copy_model_from() supports `.new_model` containing a period [BBR-CMF-
...
+  withr::defer(cleanup())
+  new_id <- "002"
+  high_id <- "0100"
+  bad_id <- "aaaaa10000aaaaa"
+  new_mod1 <- copy_model_from(MOD1, file.path(basename(LEVEL2_DIR), new_id))
+  # write a fake ctlwith a higher number
```
Comment (kyleam) [inline tests/testthat/test-copy-model-from.R:179]: `s/ctlwith/ctl with/`

**Review by kyleam — APPROVED** [review]
LGTM, just a few typo comments.

**Pull request merged**